ext/AbstractDifferentiationEnzymeExt.jl

```diff
@@ -0,0 +1,62 @@
+module AbstractDifferentiationEnzymeExt
+
+if isdefined(Base, :get_extension)
+    import AbstractDifferentiation as AD
+    using Enzyme: Enzyme
+else
+    import ..AbstractDifferentiation as AD
+    using ..Enzyme: Enzyme
+end
+
+struct Mutating{F}
+    f::F
+end
+function (f::Mutating)(y, xs...)
+    y .= f.f(xs...)
+    return y
+end
+
+AD.@primitive function value_and_pullback_function(b::AD.EnzymeReverseBackend, f, xs...)
+    y = f(xs...)
+    return y,
+    Δ -> begin
+        Δ_xs = zero.(xs)
+        dup = if y isa Real
+            if Δ isa Real
+                Enzyme.Duplicated([y], [Δ])
+            elseif Δ isa Tuple{Real}
+                Enzyme.Duplicated([y], [Δ[1]])
+            else
+                throw(ArgumentError("Unsupported cotangent type."))
+            end
+        else
+            if Δ isa AbstractArray{<:Real}
+                Enzyme.Duplicated(y, Δ)
+            elseif Δ isa Tuple{AbstractArray{<:Real}}
+                Enzyme.Duplicated(y, Δ[1])
+            else
+                throw(ArgumentError("Unsupported cotangent type."))
+            end
+        end
+        Enzyme.autodiff(
+            Enzyme.Reverse,
+            Mutating(f),
+            Enzyme.Const,
+            dup,
+            Enzyme.Duplicated.(xs, Δ_xs)...,
+        )
+        return Δ_xs
+    end
+end
+function AD.pushforward_function(::AD.EnzymeReverseBackend, f, xs...)
+    return AD.pushforward_function(AD.EnzymeForwardBackend(), f, xs...)
```

Comment:
This creates an inconsistency with the behaviour of other backends where it is guaranteed that the specified backend is used for every operation. I think the better design might be to have dedicated Reverse+Forward wrappers that allow to specify different backends for forward and reverse mode operations and pick the best mode for every call.

Reply:
I agree. This was done to make some failed tests pass which likely fail due to an Enzyme correctness issue. We should change this before merge.

Reply:
What is the test case of the correctness issue? Can you open an issue with it?